Sources/ApiVideoPlayer/View/ActionBarView.swift

```diff
@@ -0,0 +1,256 @@
+        verticalStackView.leftAnchor.constraint(equalTo: leftAnchor).isActive = true
+        verticalStackView.rightAnchor.constraint(equalTo: rightAnchor).isActive = true
+
+        actionStackView.translatesAutoresizingMaskIntoConstraints = false
+        actionStackView.leftAnchor.constraint(equalTo: bottomActionView.leftAnchor).isActive = true
+        actionStackView.topAnchor.constraint(equalTo: bottomActionView.topAnchor).isActive = true
+        actionStackView.bottomAnchor.constraint(equalTo: bottomActionView.bottomAnchor).isActive = true
+
+        // LiveButton
+        liveButton.translatesAutoresizingMaskIntoConstraints = false
+        liveButton.rightAnchor.constraint(equalTo: bottomActionView.rightAnchor).isActive = true
+        liveButton.centerYAnchor.constraint(equalTo: bottomActionView.centerYAnchor).isActive = true
+        liveButton.topAnchor.constraint(equalTo: bottomActionView.topAnchor).isActive = true
+        liveButton.bottomAnchor.constraint(equalTo: bottomActionView.bottomAnchor).isActive = true
+        liveButton.widthAnchor.constraint(equalToConstant: 40).isActive = true
+
+    }
+
+    private func updateLiveButtonColor() {
+        let remainingTime = playerController.duration - playerController.currentTime
+        if remainingTime >= CMTime(seconds: 3, preferredTimescale: 1_000) {
+            setLiveButtonTintColor(false)
+        } else {
+            setLiveButtonTintColor(true)
+        }
+    }
+
+    @objc
+    func goToLive() {
+        playerController.seek(to: playerController.duration) { completed in
+            if completed {
+                self.setLiveButtonTintColor(true)
+            } else {
+                self.setLiveButtonTintColor(false)
+            }
+        }
+    }
+
+    private func setLiveButtonTintColor(_ isRealTime: Bool) {
+        if isRealTime {
+            liveButton.tintColor = .red
+        } else {
+            liveButton.tintColor = .white
+        }
+    }
+
+    private func removeTimeObserver() {
+        if let timeObserver = timeObserver {
+            playerController.removeTimeObserver(timeObserver)
+            self.timeObserver = nil
+        }
+    }
+
+    @objc
+    private func toggleSubtitleView() {
+        delegate?.subtitleButtonTapped(subtitleButton: subtitleButton)
+    }
+}
+
+extension ActionBarView: PlayerDelegate {
+    func didPrepare() {}
+
+    func didReady() {
+        timeSliderView.duration = playerController.duration
+        if playerController.isVod {
+            liveButton.isHidden = true
+        } else if playerController.isLive {
+            liveButton.isHidden = false
+        } else {
+            print("Error: unexpected video type")
+        }
+
+        // Show or hide subtitle button
+        subtitleButton.isHidden = !playerController.hasSubtitles
+
+        // Show or hide bottomActionView if there is no button inside
+        bottomActionView.isHidden = liveButton.isHidden && subtitleButton.isHidden
```

Comment:
instead of using `subtitleButton.isHidden`, it would be better (specially for future usage) to check if `actionStackView` has at least one visible item.
Check if `UIStackView` add an API for this, otherwise, add an extension for `UIStackView` (`isEmpty` for example)